packages/app/src/components/StakeForm.tsx

```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
 interface AmountInputProps {
   className?: string;
   balance: BN | undefined;
+  amount: string;
   setAmount: Function;
 }
 
 const AmountInput: React.FC<AmountInputProps> = ({
   balance,
   className,
+  amount,
   setAmount,
-}) => (
-  <div className={className}>
-    <div className="p-8 pt-3 bg-background">
-      <div className="text-right">
-        Balance:{" "}
-        <span className="text-blue">
-          {balance ? toFixedWithPrecision(toSol(balance)) : "-"} SOL
-        </span>
+}) => {
+  const [isMax, setIsMax] = useState(false);
```

Comment:
We don't need state here, on Max click just setAmount(balance)
Keeping track if amount == balance just adds complexity with no real user benefit

Reply:
Done.